src/sentry/plugins/providers/base.py

```diff
@@ -6,13 +6,66 @@
 from rest_framework.response import Response
 from social_auth.models import UserSocialAuth
 
+from sentry.models import Integration, OrganizationIntegration
+
 from sentry.exceptions import InvalidIdentity, PluginError
 
 
 class ProviderMixin(object):
     auth_provider = None
     logger = None
 
+    def link_auth(self, user, organization, data):
```

Comment:
so this always requires a default social auth attached, did we know that its always going to be valid that this constraint is true

e.g. are there any obvious integrations that might not have coupling to a user?